GetIntoTeachingApi/Controllers/SchoolsExperience/CandidatesController.cs

```diff
@@ -57,13 +63,30 @@ public class CandidatesController : ControllerBase
                 return BadRequest(ModelState);
             }
 
+            var candidate = request.Candidate;
+
             if (appSettings.IsCrmIntegrationPaused)
             {
-                throw new InvalidOperationException("CandidatesController#SignUp - Aborting (CRM integration paused).");
+                if (_env.IsProduction)
+                {
+                    throw new InvalidOperationException("CandidatesController#SignUp - Aborting (CRM integration paused).");
+                }
+
+                // Usually, it is best practice to allow the CRM to generate sequential GUIDs which provide better
+                // SQL performance. However, in this scenario we have agreed it is beneficial to provide the GUID up-front
+                // because the School Experience app needs the Candidate ID immediately.
+                candidate.Id = Guid.NewGuid();
+
+                // This is the only way we can mock/freeze the current date/time
+                // in contract tests (there's no other way to inject it into this class).
+                request.DateTimeProvider = _dateTime;
+                string json = request.Candidate.SerializeChangeTracked();
+                _jobClient.Enqueue<UpsertCandidateJob>((x) => x.Run(json, null));
+            }
+            else
+            {
+                _upserter.Upsert(candidate);
             }
-
-            var candidate = request.Candidate;
-            _upserter.Upsert(candidate);
 
             return CreatedAtAction(
```

Comment:
This should be an `AcceptedAtAction` status 202 https://stackoverflow.com/questions/37130993/what-should-be-the-http-status-code-for-request-pending

Not sure if it matter much